src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -0,0 +1,606 @@
+            {
+                throw new ArgumentException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    Strings.WhyCommand_Error_InvalidFramework));
+            }
+        }
+
+        /// <summary>
+        /// Validates and returns the assets file for the given project.
+        /// </summary>
+        /// <param name="project">Evaluated MSBuild project</param>
+        /// <param name="logger">Logger for the 'why' command</param>
+        /// <returns>Assets file for given project</returns>
+        private static LockFile GetProjectAssetsFile(Project project, ILoggerWithColor logger)
+        {
+            if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+            {
+                logger.LogError(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.Error_NotPRProject,
+                        project.FullPath));
+
+                return null;
+            }
+
+            string assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+            if (!File.Exists(assetsPath))
+            {
+                logger.LogError(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.Error_AssetsFileNotFound,
+                        project.FullPath));
+
+                return null;
+            }
+
+            var lockFileFormat = new LockFileFormat();
+            LockFile assetsFile = lockFileFormat.Read(assetsPath);
+
+            // assets file validation
+            if (assetsFile.PackageSpec == null
+                || assetsFile.Targets == null
+                || assetsFile.Targets.Count == 0)
+            {
+                logger.LogError(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.WhyCommand_Error_CannotReadAssetsFile,
+                        assetsPath));
+
+                return null;
+            }
+
+            return assetsFile;
+        }
+
+        /// <summary>
+        /// Finds dependency graphs for a given project, and prints output to the console.
+        /// </summary>
+        /// <param name="whyCommandArgs">CLI arguments for the 'why' command</param>
+        /// <param name="msBuild">MSBuild utility</param>
+        /// <param name="project">Current project</param>
+        /// <param name="assetsFile">Assets file for current project</param>
+        private static void FindAllDependencyGraphs(
+            WhyCommandArgs whyCommandArgs,
+            MSBuildAPIUtility msBuild,
+            Project project,
+            LockFile assetsFile)
+        {
+            // get all resolved package references for a project
+            List<FrameworkPackages> frameworkPackages = msBuild.GetResolvedVersions(project, whyCommandArgs.Frameworks, assetsFile, transitive: true, includeProjectReferences: true);
```

Comment:
Instead of using this method from the MSBuildAPIUtility class, I've written a more minimal method to get the top-level package and project references (in `DependencyGraphFinder`), and this doesn't need anything other than the AssetsFile and the user input frameworks